score-vmware/src/main/java/io/cloudslang/content/vmware/utils/FindObjects.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * ******************************************************
+ * Copyright VMware, Inc. 2014. All Rights Reserved.
+ * ******************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudslang.content.vmware.utils;
+
+import com.vmware.vim25.*;
+import io.cloudslang.content.vmware.constants.Constants;
+
+import java.util.ArrayList;
+import java.util.List;
+/*
+ * This class includes several "findObject" methods.  You will find very
+ * similar, and in some cases identical methods in the VMwareConnection class.
+ * Why the duplication? Because the two classes have different interface
+ * philosophies.  This class has a library style interface: it is a
+ * connection of static methods, and they use "raw" (unencapsulated) data
+ * from a connection. VMwareConnection has an object oriented design and
+ * encapsulates the connection.
+ *
+ * No attempt has been made to "factor out" repetitive code, especially from
+ * the different versions of findObject.  This was done (or not-done) deliberately
+ * in order to make it easier to follow the code, for learning purposes.
+ */
+
+/**
+ * Utility methods for working with Java Objects in VMware's vCenter Java libraries. These methods
+ * work mostly with Managed Object Reference objects and Object Content objects.
+ */
+
+public class FindObjects {
```

Comment:
if the utility class contains only static methods, it is a good idea to declare it static class or final + private constructor